src/components/document/use-publish-dialog.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import PublishSvg from "../../assets/icons/publish/publish-icon-default.svg";
+import { useCustomModal } from "../../hooks/use-custom-modal";
+import { useDBStore, useProblemPath, useStores, useUIStore, useUserStore } from "../../hooks/use-stores";
+import { DocumentModelType } from "../../models/document/document";
+import { ProblemDocument } from "../../models/document/document-types";
+
+import "./publish-dialog.scss";
+
+export const usePublishDialog = (document: DocumentModelType) => {
+  const stores = useStores();
+  const user = useUserStore();
+  const db = useDBStore();
+  const problemPath = useProblemPath();
+  const ui = useUIStore();
+  const docTypeString = document.getLabel(stores.appConfig, 1);
+  const docTypeStringL = document.getLabel(stores.appConfig, 1, true);
+
+  const getProblemBaseTitle = (problemTitle: string) => {
+    const match = /[\d.]*[\s]*(.+)/.exec(problemTitle);
+    return match && match[1] ? match[1] : problemTitle;
```

Comment:
```suggestion
    return match?.[1] || problemTitle;
```
I know this is existing code that just moved, but since we're touching it we might as well adopt modern simplifications.